pkg/kubelet/apis/config/types.go

```diff
@@ -480,6 +480,16 @@ type KubeletConfiguration struct {
 	// If not specified the default value is ContainerRuntimeEndpoint
 	// +optional
 	ImageServiceEndpoint string
+
+	// MaxCheckpointsPerContainer specifies the maximum number of checkpoints
+	// that Kubernetes will create of one specific container before removing old
+	// checkpoints. This option exist to ensure the local disk is not filled
+	// with container checkpoints. This option is per container. The field
+	// value must be greater than 0.
```

Comment:
The size of a checkpoint depends on the runtime state of processes running in a container. It can be a few megabytes for some applications, and several gigabytes for others. In addition, the size can vary from time to time as applications allocate/deallocate memory, create and delete files, etc.

Would it make sense to use a "quota" that the specifies total maximum size for all checkpoints of a container, and remove the oldest checkpoints only when the quota has been exceeded?

Reply:
Definitely a good idea. But maybe it should be a separate PR.

Reply:
> put another way.. why was 10 selected and not 1?

No real reason. I just selected a number which felt right for me. Anything but unlimited is an improvement from my point of view.